src/ts/Detail/DetailForm.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+  DatePicker,
+  FontIcon,
+  SelectField,
+  MenuItem,
+} from 'material-ui';
+import debug from 'debug';
+const log = debug('app:DetailForm');
+
+export default class DetailForm extends React.Component<any, any> {
+  private offer;
+
+  constructor(props) {
+    super(props);
+    const offer = props.offer;
+    this.offer = offer;
+    const price = offer.price.unit[Object.keys(offer.price.unit)[0]];
+    const now = new Date();
+    log(price);
+    this.state = {
+      price,
+      startDate: now,
+      endDate: new Date(now.getTime() + 3 * 24 * 60 * 60 * 1000),
+      equipment: 1,
+      total: parseInt(price, 10) * 3,
+    };
+    this.endDateChange = this.endDateChange.bind(this);
+    this.startDateChange = this.startDateChange.bind(this);
+    this.priceChange = this.priceChange.bind(this);
+    this.equipmentChange = this.equipmentChange.bind(this);
+  }
+
+  recalculate() {
+    const start = new Date(this.state.startDate);
+    const end = new Date(this.state.endDate);
+    const total = parseInt(this.state.price, 10) * Math.floor((end.getTime() - start.getTime()) / (1000 * 60 * 60 * 24));
+    this.setState({ total });
+  }
+
+  startDateChange(ev, date) {
+    log('start date changed', date);
+    this.setState({ startDate: date });
+    this.recalculate();
+  }
+
+  endDateChange(ev, date) {
+    log('start date changed', date);
+    this.setState({ endDate: date });
+    this.recalculate();
+  }
+
+  equipmentChange(ev, index, equipment) {
+    log('equipment changed', equipment);
+    this.setState({ equipment });
+  }
+
+  priceChange(ev, index, value) {
+    log('price changed', value);
+    this.setState({ price: value });
+    this.recalculate();
+  }
+
+  render() {
+    const {
+      price,
+    } = this.offer;
+
+    const selectPrice = _.map(price.unit, (num, key) => (
+      <MenuItem
+        key={num}
+        value={num}
+        primaryText={`Base price: ${num} ${price.value} / ${key}`}
+      >
+      </MenuItem>));
+
+    return (
+      <div>
+        <div className="field">
+          <FontIcon className="fa fa-money" />
+          <SelectField
```

Comment:
in particular, in JSX / TSX, we should not keep the space with the brackets. Please check for inline. Otherwise, indent.